docs/js-dev-standards.md

```diff
@@ -0,0 +1,101 @@
+# JavaScript Software Development Standards
+
+The goal of this document is to describe how our team’s projects are typically structured. This includes normative tooling and structure for different kinds of software (e.g. libraries vs applications), and what typical flows a developer should expect to be available.
+
+
+## Goals and Principles
+* __Consist__ workflow across projects
+* __Familiar__ to developers, and easy to adopt
+* __Fast__ CI/CD and development/testing workflows. To support this principle, we transpile and bundle code _as little as possible_ and support dev-loops that only use source code.
+* __Modern__ We lean into the latest features of the JavaScript ecosystem, including TypeScript, ESM, and modern language features.
+
+## Project Structure
+Our projects are structured as monorepos using Yarn 3+ (or whatever is latest). Our software is usually composed of many moving parts: libraries, demo applications, full-blown applications, test suites, and more. To manage these in a single repository, we use Yarn workspaces.
+
+### JavaScript-Only Monorepos
+For single-language monorepos, workspaces are created under the `packages` folder. For example, a monorepo with a library and a demo application would look like this:
+
+```text
+- packages
+	- my-library
+		- package.json
+	- my-demo
+		- package.json
+```
+
+### Multi-Language Monorepos
+When authoring monorepos with multiple languages, Yarn workspaces are created under the `javascript` folder, and other languages would follow a similar pattern. For example, a monorepo with a library, demo application, and a Go backend would look like this:
+
+```text
+- javascript
+  - my-library
+		- package.json
+	- my-demo
+		- package.json
+- go
+	- my-backend
+		- go.mod
+		- go.sum
+		- main.go
+```
+
+The rest of this document will focus primarily on JavaScript.
+
+## Common Build Tasks
+The following top-level tasks expected to exist in all JavaScript projects. 
+
+* __clean__ - this task cleans the project, removing any artifacts or temporary files.
+* __release__ - this task applies semver changes, executes build verification, and publishes artifacts to the package registry.
+* __ci__ - this task is used to chain together an end-to-end continuous verification pipeline. It should run all tests, linting, and other verification tasks.
+* __start__ - this task starts any webapps or servers in the project in development mode.
+* __build__ - this task builds artifacts and performs typechecking for library packages within the project.
+* __test__ - this task runs all tests in the project.
+* __check__ - this task checks code for linting and formatting issues.
+* __fix__ - this task checks code for linting and formatting issues and attempts to fix as many as possible.
+* __update_sdks__ this is a utility command that's occasionally used to update yarn sdks (e.g. `yarn dlx @yarnpkg/sdks vscode`)
+
+## CI Tasks
```

Comment:
I'm not sure I follow, I don't think we ever trigger actions manually.

Reply:
Right, correction: I think we do this internally with DevOps pipelines (e.g., ShowWhy prod)

Reply:
Yep we do that with Showwhy prod, but since we use a different branch and it's more like a whole manual and internal process, I don't think we need to worry about these.